frappe/public/js/frappe/router.js

```diff
@@ -170,8 +170,13 @@ frappe.router = {
 		} else {
 			standard_route = ['List', doctype_route.doctype, frappe.utils.to_title_case(route[2])];
 			if (route[3]) {
-				// calendar / kanban / dashboard / folder name
-				standard_route.push([...route].splice(3, route.length));
+				if (Array.isArray(route)) {
+					// folder
+					standard_route.push(...route.splice(3, route.length));
+				} else {
+					// calendar / kanban / dashboard name
+					standard_route.push(route[3]);
```

Comment:
if the route is not an array, shouldn't we push the complete route?
```suggestion
					standard_route.push(route]);
```

Reply:
we only need third value other preceding values are already in there in standard_route.

Reply:
@shariquerik can you share an example of this case?

I think if `route` is a string like "document-222" the value of `route[3]` will be a single character i.e, "u"... what am I missing 🤔

Reply:
@surajshetty3416 I think it's either Array or Array-like (`arguments`). Or at least it's supposed to be.

Reply:
Suggestion from Sagar resolves this issue.